controllers/disk.go

```diff
@@ -42,118 +42,251 @@ func DiskCreate(c *gin.Context) {
-			UUID:         disk.GetUUID().String(),
-			Name:         requestBody.Name,
-			ProviderUUID: providerUUID.String(),
-			Link:         authCode,
-		},
+		EmptySuccessResponse: *responses.CreateEmptySuccessResponse(_disk),
+		Link:                 authCode,
 	})
 }
 
 func DiskOAuth(c *gin.Context) {
 	var requestBody requests.OAuthRequest
+	var _diskUUID string
+	var diskUUID uuid.UUID
+	var err error
+	var _disk dbo.Disk
 
 	// Retrieve and validate data from request
-	if err := c.ShouldBindJSON(&requestBody); err != nil {
+	if err = c.ShouldBindJSON(&requestBody); err != nil {
 		c.JSON(422, responses.NewValidationErrorResponse(err))
 		return
 	}
 
 	userData, _ := c.Get("UserData")
 	userUUID := userData.(middleware.UserData).UserUUID
 
-	volumeUUID, err := uuid.Parse(requestBody.VolumeUUID)
-	if err != nil {
-		c.JSON(422, responses.NewValidationErrorResponseSingle(constants.VAL_UUID_INVALID, "VolumeUUID", "Provided VolumeUUID is not a valid UUID"))
-		return
-	}
-
-	diskUUID, err := uuid.Parse(requestBody.DiskUUID)
-	if err != nil {
-		c.JSON(422, responses.NewValidationErrorResponseSingle(constants.VAL_UUID_INVALID, "DiskUUID", "Provided DiskUUID is not a valid UUID"))
-		return
-	}
+	_diskUUID = c.Param("DiskUUID")
+	diskUUID, err = uuid.Parse(_diskUUID)
 
-	providerUUID, err := uuid.Parse(requestBody.ProviderUUID)
+	err = db.DB.DatabaseHandle.Where("uuid = ?", _diskUUID).Preload("Provider").Preload("Volume").Find(&_disk).Error
 	if err != nil {
-		c.JSON(422, responses.NewValidationErrorResponseSingle(constants.VAL_UUID_INVALID, "ProviderUUID", "Provided ProviderUUID is not a valid UUID"))
-		return
+		c.JSON(404, responses.NewNotFoundErrorResponse(constants.DATABASE_DISK_NOT_FOUND, "Cannot find a disk with the provided UUID"))
 	}
 
-	volume := models.Transport.GetVolume(userUUID, volumeUUID)
+	volume := models.Transport.GetVolume(userUUID, _disk.VolumeUUID)
 	if volume == nil {
-		c.JSON(404, responses.NewNotFoundErrorResponse(constants.TRANSPORT_VOLUME_NOT_FOUND, "Cannot find volume with provided UUID"))
+		c.JSON(404, responses.NewNotFoundErrorResponse(constants.TRANSPORT_VOLUME_NOT_FOUND, "Cannot find a volume with the provided UUID"))
 		return
 	}
 
-	disk := (volume.GetDisk(diskUUID)).(disk2.OAuthDisk)
+	disk := (volume.GetDisk(diskUUID)).(OAuthDisk.OAuthDisk)
 	if disk == nil {
-		c.JSON(404, responses.NewNotFoundErrorResponse(constants.TRANSPORT_DISK_NOT_FOUND, "Cannot find disk with provided UUID or disk is not associated with provided volume"))
+		c.JSON(404, responses.NewNotFoundErrorResponse(constants.TRANSPORT_DISK_NOT_FOUND, "The provided disk is not associated with the provided volume"))
 		return
 	}
 
 	config := disk.GetConfig()
 	config.Endpoint.AuthStyle = oauth2.AuthStyleInParams
 	tok, err := config.Exchange(c, requestBody.Code)
 	if err != nil {
+		c.JSON(422, responses.NewValidationErrorResponse(err, "Could not retrieve the oauth token"))
 		log.Fatalf("Unable to retrieve token from web %v", err)
```

Comment:
Czy tutaj nie brakuje return?

Reply:
Fatalf powoduje panic() (generalnie os.Exit()) ale to jest relikt

zamieniam to na return dla zachowania 'konsystencji' xd